ad7768-1fmcz/ad77681.c

```diff
@@ -282,13 +665,291 @@ int32_t ad77681_set_power_mode(struct ad77681_dev *dev,
 int32_t ad77681_set_mclk_div(struct ad77681_dev *dev,
 			     enum ad77681_mclk_div clk_div)
 {
-	ad77681_spi_write_mask(dev,
+	int32_t ret = 0;
+
+	ret |= ad77681_spi_write_mask(dev,
 			       AD77681_REG_POWER_CLOCK,
 			       AD77681_POWER_CLK_MCLK_DIV_MSK,
 			       AD77681_POWER_CLK_MCLK_DIV(clk_div));
 
 	dev->mclk_div = clk_div;
 
+	return ret;
+}
+
+/**
+ * Set the VCM output.
+ * @param dev - The device structure.
+ * @param VCM_out - The VCM output voltage.
+ * 					Accepted values: AD77681_VCM_HALF_VCC
+ *									 AD77681_VCM_2_5V
+	  								 AD77681_VCM_2_05V
+									 AD77681_VCM_1_9V
+									 AD77681_VCM_1_65V
+									 AD77681_VCM_1_1V
+									 AD77681_VCM_0_9V
+									 AD77681_VCM_OFF
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_set_VCM_output(struct ad77681_dev *dev,
+	enum ad77681_VCM_out VCM_out)
+{
+	int32_t ret;
+	
+	ret = ad77681_spi_write_mask(dev,
+		AD77681_REG_ANALOG2,
+		AD77681_ANALOG2_VCM_MSK,
+		AD77681_ANALOG2_VCM(VCM_out));
+
+	dev->VCM_out = VCM_out;
```

Comment:
Check return value, before doing anything else.
Check in code for similar issues.